tests/ga/test_extension.py

```diff
@@ -605,6 +605,50 @@ def test_ext_handler(self, *args):
 
         self._assert_no_handler_status(protocol.report_vm_status)
 
+    def test_it_should_zip_waagent_status_when_incarnation_changes(self, *args):
+        """
+        This test checks when the incarnation changes the waagnet_status file for the previous incarnation
+        is added into the history folder for the previous incarnation and gets zipped
+        """
+        temp_files = [
+            'ExtensionsConfig.1.xml',
+            'GoalState.1.xml',
+            'OSTCExtensions.ExampleHandlerLinux.1.manifest.xml',
+            'waagent_status.1.json'
+        ]
+
+        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)  # pylint: disable=no-value-for-parameter
```

Comment:
new code should not have pylint suppressions unless there is no way to avoid them (and it that case, the reason should be documented)

Reply:
Addressed in the next commit